build_tools/benchmarks/upload_benchmarks_to_dashboard.py

```diff
@@ -291,40 +293,62 @@ def check_file_path(path):
 
 def main(args):
   # Collect benchmark results from all files.
-  all_results = []
-  for benchmark_file in args.benchmark_files:
+  benchmark_results = []
+  statistic_results = []
+  for benchmark_file in args.result_files:
     with open(benchmark_file) as f:
       content = f.read()
-    all_results.append(BenchmarkResults.from_json_str(content))
-  for other_results in all_results[1:]:
-    all_results[0].merge(other_results)
-  all_results = all_results[0]
+    is_statistics = "statistics" in json.loads(content)
+    if is_statistics:
+      statistic_results.append(StatisticResults.from_json_str(content))
+    else:
+      benchmark_results.append(BenchmarkResults.from_json_str(content))
+  for other_results in benchmark_results[1:]:
+    benchmark_results[0].merge(other_results)
+  benchmark_results = benchmark_results[0]
+  for other_results in statistic_results[1:]:
+    statistic_results[0].merge(other_results)
+  statistic_results = statistic_results[0]
 
   # Register a new build for the current commit.
-  commit_hash = get_git_commit_hash(all_results.commit, verbose=args.verbose)
+  commit_hash = get_git_commit_hash(benchmark_results.commit,
+                                    verbose=args.verbose)
   commit_count = get_git_total_commit_count(commit_hash, verbose=args.verbose)
   add_new_iree_build(commit_count,
                      commit_hash,
                      dry_run=args.dry_run,
                      verbose=args.verbose)
 
-  # Get the mean time for all benchmarks.
   aggregate_results = {}
-  for benchmark_index in range(len(all_results.benchmarks)):
-    benchmark_case = all_results.benchmarks[benchmark_index]
+
+  # Get the mean time for all benchmarks.
+  for benchmark_index in range(len(benchmark_results.benchmarks)):
+    benchmark_case = benchmark_results.benchmarks[benchmark_index]
     benchmark_info = benchmark_case.benchmark_info
 
     # Make sure each benchmark has a unique name.
     name = str(benchmark_info)
     if name in aggregate_results:
       raise ValueError(f"Duplicated benchmarks: {name}")
 
-    mean_time = all_results.get_aggregate_time(benchmark_index, "mean")
+    mean_time = benchmark_results.get_aggregate_time(benchmark_index, "mean")
     aggregate_results[name] = (mean_time, benchmark_info)
 
+  # Get the value for all statistics.
+  for statistic_index in range(len(statistic_results.statistics)):
+    statistic_case = statistic_results.statistics[statistic_index]
+    statistic_info = statistic_case.statistic_info
```

Comment:
If `statistic_results.statistics` is a list / typical iterable data type, can't you simplify this?

```suggestion
  for statistic_case in statistic_results.statistics:
    statistic_info = statistic_case.statistic_info
```

same above with 

```python
for benchmark_index in range(len(benchmark_results.benchmarks)):
    benchmark_case = benchmark_results.benchmarks[benchmark_index]
```

if you need the index, the pythonic way to iterate is https://stackoverflow.com/a/522578

```python
for idx, val in enumerate(ints):
    print(idx, val)
```